src/xml/error.cr

```diff
@@ -11,19 +11,36 @@ class XML::Error < Exception
     super(message)
   end
 
-  @@errors = [] of self
+  @@max_error_capacity = 5
+
+  @[Deprecated("This class property is deprecated. XML errors are accessible directly in the respective context via `XML::Reader#errors` and `XML::Node#errors`.")]
+  def self.max_error_capacity=(@@max_error_capacity)
+  end
+
+  @[Deprecated("This class property is deprecated. XML errors are accessible directly in the respective context via `XML::Reader#errors` and `XML::Node#errors`.")]
+  def self.max_error_capacity
+    @@max_error_capacity
+  end
+
+  @@errors = Deque(self).new(max_error_capacity)
```

Comment:
**Question**: this initializer will run before user code, won't `@max_error_capacity` always be the default 5, whatever I set?

Reply:
Argh, yeah it will 🤦